### ISSUE TITLE
Annotations

### ISSUE THREAD

**Issue by mads-hartmann** [tree-sitter/tree-sitter-scala#2]
Something like the following: 

```
==================================
Annotation Types
==================================

class B[@scala.specialized[B] c]

---

(compilation_unit
  (class_definition
    (identifier)
    (type_parameters
      (annotation
        (simple_type
          (simple_type (stable_identifier (identifier) (identifier)))
          (type_arguments (type_identifier))))
      (identifier))))
```

**Comment by maxbrunsfeld** [comment]
Yeah, this seems like one of the biggest pieces of Scala's syntax that we don't yet handle.

**Comment by gemelen** [comment]
Implemented by #13
